### ISSUE TITLE
The .ftml file won't open as FoldingText outline but as HTML in Windows

### ISSUE THREAD

**Issue by juranta** [FoldingText/foldingtext-for-atom#44]
In my work Windows, I can select "New Outline" and it will open a file in FT outline mode. After I save the outline to disk with .ftml extension and close the tab, the next time I open the outline, it will not open in FT outline mode but in HTML mode, showing the FTML HTML syntax instead of the outline. I'm not certain, but it may have started with the 0.3 update. It works properly on my Macs.

**Comment by jessegrosjean** [comment]
Are you sure when you save that it has a `ftml` file extension?

Assuming it does I'm not sure one a next step. I guess:
1. Make sure your version of Atom is the latest 0.204.0
2. Make sure you have the latest FoldingText for Atom 0.4.1

If it still doesn't work let me know... I don't have windows which makes debugging this a bit tricky. Do you know how to use the Javascript debugger? :\

**Comment by juranta** [comment]
I am sure the file has a `ftml` extension. I had used the same file for some days, then it broke, I think it was with the 0.3 update. I tried removing and installing again the plugin, I also tried creating new files with the `.ftml` extension. I tried moving the entire `.atom` directory away (with the versions listed below), then restarting Atom, etc, but it still didn't work.

I'm currently working as a software developer in a quite "polyglot" environment. I'm happy to try debugging if it helps. 
- Versions
  - Version of Atom
    - 0.204.0
  - Version of FoldingText
    - 0.4.1

**Comment by jessegrosjean** [comment]
Ok, must be something different about windows maybe. To debug you can try:
1. View > Developer > Toggle Dev Tools
2. In the dev tools navigate to the "Sources" section.
3. Then you want to go to the "foldingtext.coffee" source code file. On OS X I do <kbd>Command-O</kbd> and type "foldingtext.coffee" to navigate to it. I guess you have no Command key on windows, but maybe try whatever the equivalent is.
4. Then in that file look for `atom.workspace.addOpener`. Should be called twice, put a breakpoint in the second longer version. On my computer that would be line 98-146. After putting in the breakpoint try opening a .ftml file and then step through that functions logic. It should be that and outline editor is opened and returned.

**Comment by juranta** [comment]
I went to that function and added some debug `console.log` printings. That function does get executed. But it seems that on Windows, the value of `parsedUURL.protocol` is actually `c:`. That `c:` is a Windows drive letter, if you haven't ever used Windows. ;) Well, it's actually `C:\` but languages often change it to `c:/` or accept it. Then you have an if statement:

```
if (parsedURL.protocol === 'file:' || !parsedURL.protocol) {
```

It doesn't go to that branch because `parsedURL.protocol` is actually `c:`.

**Comment by jessegrosjean** [comment]
Thanks for looking... can you tell me what the value of the passed in `filePath` is?

**Comment by juranta** [comment]
When I did a `console.log` for file path, it said `C:\users\myname\Notes\my_outline.ftml`. I haven't used much Javascript lately, so I didn't remember straight hand how that string is generated or how to get more useful info out of it in Javascript. (Or CoffeeScript).

**Comment by jessegrosjean** [comment]
Thanks, that's what I was looking for. I was wondering if I could detect windows file paths by the slash direction, looks like I can.

**Comment by jessegrosjean** [comment]
@juranta I've just pushed v0.4.2. Can you try it and let me know if it fixes the problem?

**Comment by juranta** [comment]
I tried it, but unfortunately the `protocol` variable is still `c:` just before the test, and it does not work. It's sleeping time for me now for me, but I can try to see further to it with more time.

**Comment by juranta** [comment]
Note that there is no character `\` at the variable `protocol` anymore.

**Comment by juranta** [comment]
I did a bit more debugging. Now it does pass that if test and go inside the block. But later in that block, there's this part:

```
decodedURLPath = decodeURI(parsedURL.pathname)
```

In my case, I happened to have a #tag in my file name, as in `my file with #tag #tag2.ftml`. That decodedURLPath stopped at # character so that it had the value `my file with`. Actually `parsedURL.pathname` is already cut from that character. And then, when it tried to get mime type, it returned undefined and that ended the block.

When I tried to create a new outline with a simple name `test.ftml`, it started some infinite loop which stuck Atom. I was watching as it was printing debug messages from this same function during the loop. However, I didn't yet find out where it was stuck in. The debug `console.log`s had disappeared when I restarted Atom. 

Is there perhaps a way to change in the case of Windows `url.parse(filePath, true)` to something that will create the same object from Windows filePath, which is not an url?
